### PR TITLE
DX: add spelling checker and improve texts

### PR DESCRIPTION
**Pull request by redeboer**
- Enforced [cSpell](https://cspell.org) as spelling checker through pre-commit and vscode.
- Fixed typos.
- Reformulated some of the paragraphs.

### CODE REVIEW

**Comment by redeboer** [comment]
This PR is meant to further clarify some of the pages, while implementing a serializer for [ampform-dpd](https://compwa.github.io/ampform-dpd). Will post comments here if things are unclear to me and we can improve the format description if needed.

File: docs/description.md [hunk 1]
```diff
@@ -164,26 +173,33 @@ Vertices define the nodes in the decay graphs, where one particle transits into
...
-  \end{align}
-  $$
-
-  The `helicity` indicated no recoupling, the factor is $1$ for a pair of selected helicities ($\lambda_a^0$ and $\lambda_b^0$) and zero for other combinations. The `ParityRecoupling` is controlled by the controlled by the `parity factor`, $f$, and gives non-zero value for two combination of the helicity pair, the selected one and the opposite. Finally, the `RecouplingLS` computes the value of the recoupling functions from Clebsch–Gordan coefficients. For spin-half particles, this recoupling is equivalent to `ParityRecoupling`, with $f = (-1)^l$.
+  These reflect different ways of relating combinations of the helicity indices to a real-valued "recoupling coefficient".
+
+  - `helicity` indicates no recoupling: the factor is $1$ for a pair of selected helicities ($\lambda_a^0$ and $\lambda_b^0$) and zero for other combinations.
+    $$
+    H^\text{helicity}(\lambda_a,\lambda_b|\lambda_a^0,\lambda_b^0) = \delta_{\lambda_a,\lambda_a^0}\delta_{\lambda_b,\lambda_b^0}
+    $$
```
Comment (redeboer) [inline docs/description.md:183]: I find this part to be confusing. @mmikhasenko, what are $\lambda_a$ and $\lambda_b$ (without the $^0$)?

File: docs/description.md [hunk 2]
```diff
@@ -164,26 +173,33 @@ Vertices define the nodes in the decay graphs, where one particle transits into
...
+  - `ls` computes the value of the recoupling functions from Clebsch–Gordan coefficients.
+    $$
+    \begin{multline}
+    H^\text{ls}(\lambda_a,\lambda_b|l,s,j_a,j_b,j) = \\
+      \sqrt{\frac{2l+1}{2j+1}}
+      \left\langle j_a,\lambda_a; j_b,-\lambda_b|s,\lambda_a-\lambda_b\right\rangle
+      \left\langle l,0; s,\lambda_a-\lambda_b|j,\lambda_a-\lambda_b\right\rangle
+    \end{multline}
+    $$
+    For spin-half particles, this recoupling is equivalent to the `parity` recoupling, with $f = (-1)^l$.
```
Comment (redeboer) [inline docs/description.md:199]: Still missing:
- `helicities`
- `formfactor` (optional) is still missing here.
  As a side note, I would write `form_factor` because formfactor is not a word.

File: docs/description.md [hunk 3]
```diff
@@ -144,11 +157,15 @@ A &= n_{j_0} D_{\tau, \lambda_2}^{j_0}(\text{angles}_{[[3,1],4]}) \\
 \end{align}
 $$
 
-- $D_{\tau, \lambda_2}^{j_0}(\text{angles}_{[[3,1],4]})$: This function describes the transformation for particle 2's helicity (`λ2`) in the overall rest frame of the system (comprising particles 3, 1, 4, and 2). Here, particle 2's helicity is defined relative to the frame where all other particles are considered, emphasizing its position in the decay sequence.
+- $D_{\tau, \lambda_2}^{j_0}(\text{angles}_{[[3,1],4]})$ describes the transformation for the helicity of particle&nbsp;`2`, $\lambda_2$, in the overall rest frame of the system (comprising particles `3`, `1`, `4`, and `2`).
+  Here, $\lambda_2$ is defined relative to the frame where all other particles are considered, emphasizing its position in the decay sequence.
```
Comment (redeboer) [inline docs/description.md:161]: We need to explain what $\tau$ is here.

File: docs/description.md [hunk 4]
```diff
@@ -144,11 +157,15 @@ A &= n_{j_0} D_{\tau, \lambda_2}^{j_0}(\text{angles}_{[[3,1],4]}) \\
...
 
-- $D_{\nu, \lambda_4}^{j_{[[3,1],4]}}(\text{angles}_{[3,1]})$: For particle 4, its helicity (`λ4`) is defined within the rest frame of the (3,1,4) system. This frame is obtained from the overall rest frame by applying a rotation and boost, signifying the progression of the decay sequence and the specific frame where particle 4's helicity is defined.
+- $D_{\nu, \lambda_4}^{j_{[[3,1],4]}}(\text{angles}_{[3,1]})$: For particle `4`, its helicity, $\lambda_4$, is defined within the rest frame of the `[3,1,4]` system.
+  This frame is obtained from the overall rest frame by applying a rotation and boost, signifying the progression of the decay sequence and the specific frame where particle `4`'s helicity is defined.
```
Comment (redeboer) [inline docs/description.md:165]: We need to explain what $\nu$ means.

File: docs/description.md (hunk 3 above)
Comment (redeboer) [reply docs/description.md:161]: > emphasizing its position in the decay sequence

This addition makes me doubt what the point of the sentence is

File: docs/description.md [hunk 5]
```diff
@@ -144,11 +157,15 @@ A &= n_{j_0} D_{\tau, \lambda_2}^{j_0}(\text{angles}_{[[3,1],4]}) \\
 \end{align}
 $$
 
-- $D_{\tau, \lambda_2}^{j_0}(\text{angles}_{[[3,1],4]})$: This function describes the transformation for particle 2's helicity (`λ2`) in the overall rest frame of the system (comprising particles 3, 1, 4, and 2). Here, particle 2's helicity is defined relative to the frame where all other particles are considered, emphasizing its position in the decay sequence.
+- $D_{\tau, \lambda_2}^{j_0}(\text{angles}_{[[3,1],4]})$ describes the transformation for the helicity of particle&nbsp;`2`, $\lambda_2$, in the overall rest frame of the system (comprising particles `3`, `1`, `4`, and `2`).
```
Comment (redeboer) [inline docs/description.md:160]: > overall rest frame of the system (comprising particles `3`, `1`, `4`, and `2`)

Maybe "overall rest frame of the subsystem that comprises particles `3`, `1`, `4`, and `2`?"?

But why does it include `2`?

File: docs/description.md [hunk 6]
```diff
@@ -126,15 +135,19 @@ It species the main properties such as spin, and masses of all particles.
...
 
-In the context of the conventional helicity formalism, the `reference_topology` array implicitly prescribes a method for defining helicities. It comes from the specification of a default quantization frame for each stage of the decay process. The helicity values employed in the indices of Wigner rotations `D_{λ1, λ2}` and couplings `H_{λ1, λ2}` are thus indicative of this frame. When considering a particle's helicity in any other frame, it must be treated as a superposition of the states defined by the default quantization.
+In the context of the conventional helicity formalism, the `reference_topology` array implicitly prescribes a method for defining helicities, as it can be used to define a default quantization frame for each stage (decay node) in the decay process.
+When considering the helicity of a particle in any other frame, it must be treated as a superposition of the states with regard to the default quantization.
+The helicity values employed in the indices of Wigner rotations `D_{λ1, λ2}` and couplings `H_{λ1, λ2}` are thus indicative of this frame.
```
Comment (redeboer) [inline docs/description.md:142]: Hard to follow this sentence.

File: docs/description.md (hunk 1 above)
Comment (mmikhasenko) [reply docs/description.md:191]: Any helicity coupling can be evaluated as a function of helicities, λa and λb. Think of `ls` which is non-zero for several values if the (λa, λb) pairs

File: docs/description.md (hunk 2 above)
Comment (mmikhasenko) [reply docs/description.md:207]: agree for renaming, should we do it in this PR?

Comment (redeboer) [reply docs/description.md:207]: I'll do that in a separate PR then

File: docs/description.md (hunk 6 above)
Comment (mmikhasenko) [reply docs/description.md:142]: raphrasing now

File: docs/description.md (hunk 5 above)
Comment (mmikhasenko) [reply docs/description.md:160]: the overall center of momentum includes all.
I'm rephrasing, should read better soon.

**Review by mmikhasenko — COMMENTED** [review]
read thought test and comment, minor rephrasing

**Comment by mmikhasenko** [comment]
#48 is duplicating this PR
